torch/autograd/gradcheck.py

```diff
@@ -1019,46 +1004,45 @@ def gradcheck(
     Returns:
         True if all differences satisfy allclose condition
     """
-    def fail_test(msg):
-        if raise_exception:
-            raise RuntimeError(msg)
-        return False
-
-    tupled_inputs = _as_tuple(inputs)
+    # This is just a wrapper that handles the raise_exception logic
+    args = locals()
+    args.pop("raise_exception")
```

Comment:
Does that modify the actual `locals` dict inplace? Or is it a shallow copy that you modify?

Reply:
We are actually modifying the locals dict inplace. At first that seemed okay, but 
from https://nedbatchelder.com/blog/201211/tricky_locals.html it looks like there are cases where the local symbol table is only copied into locals() after every statement. Good catch!